setup.py

```diff
@@ -1,26 +1,18 @@
 #!/usr/bin/env python
 
+import glob
 import sys
 import os
 import warnings
 
+from setuptools import setup, Extension
 from distutils.dist import Distribution
 
 display_option_names = Distribution.display_option_names + ['help', 'help-commands']
 query_only = any('--' + opt in sys.argv for opt in display_option_names) or len(sys.argv) < 2 or sys.argv[1] == 'egg_info'
 
-# Use setuptools for querying the package, normal builds use distutils
```

Comment:
I needed it to work for some commands only available in ``setuptools`` and not ``distutils``, for example ``python setup.py egg_info``.  But more importantly, I wanted to be able to run "query only" commands like ``python setup.py --help`` without having import errors if ``numpy`` and ``cython`` weren't available.

I'd love to stop using ``setuptools`` and just use ``distutils`` if we can refactor the ``setup.py`` to still allow ``--help`` with no numpy or cython.